advisor-backend/routes/schedule.js

```diff
@@ -208,6 +292,47 @@ const generateSchedule = async (
   return years;
 };
 
+const moveCourseToDifferentYear = (
+  schedule,
+  desiredYear,
+  originalYear,
+  course
+) => {
+  for (let i = 0; i < schedule.length; i++) {
+    const year = schedule[i];
+    // delete it from this year
+    if (year.number === originalYear) {
+      // for both semesters
+      year.semesters[0].classes = year.semesters[0].classes.filter(
+        (classItem) => classItem.id !== course.id
+      );
```

Comment:
You could probably tighten this up slightly by using a Set like you do in a few other places.